Update Docs For Bound Audience Explanation (#30519)
hashicorp/vault: website/content/api-docs/auth/jwt.mdx

- `name` `(string: <required>)` - Name of the role.
- `role_type` `(string: <optional>)` - Type of role, either "oidc" (default) or "jwt".
- `bound_audiences` `(array: <optional>)` - List of `aud` claims to match against.

>   This parameter is **mandatory** for "jwt" roles if the JWT contains an audience (which is more often than not the case),
>   and **must** match at least one of the JWT's associated `aud` claims. Optional for "oidc" roles.

Reviewer: ```suggestion
  The `bound_audiences` parameter is required for "jwt" roles that contain an
  audience (typical case) and **must** match at least one of the associated JWT
  `aud` claims.
```
Style correction: avoid "this" as a pronoun, write with simple, declarative sentences.

I'm assuming the fact that this is the typical case is the important part since it's repeated elsewhere so I left it in, but typically I'd recommend removing it for simplicity.
